reconcile/utils/ocm_base_client.py

```diff
@@ -161,8 +162,15 @@ def access_token_client_secret(self) -> HasSecret:
         ...
 
 
+class OCMAPIClientConfiguration(BaseModel, arbitrary_types_allowed=True):
```

Comment:
why do we allow arbitrary types here?

Reply:
hmm. good point. i think at one point mypy was complaining and this fixed it but now it works without as well. going to remove it

Reply:
ah now i see. i get runtime errors without this when i provide a `VaultSecret` for the `HasSecret`